executorCwl/src/main/scala/dx/executor/cwl/CwlWorkflowExecutor.scala

```diff
@@ -446,13 +446,14 @@ case class CwlWorkflowExecutor(workflow: Workflow,
     }
 
     private def evaluateCallStepInputs(
-        isConditinal: Boolean = false
+        inputs: Option[Map[DxName, (WorkflowStepInput, (CwlType, CwlValue))]] = None,
+        isConditional: Boolean = false
     ): Map[DxName, (WorkflowStepInput, (CwlType, CwlValue))] = {
       // Evaluate all the step inputs for a call. There may be step inputs that are not passed to
       // the callee but are referred to in expressions of other step inputs' valueFrom fields. Note
       // that the step input's type here might not match the callee's input type if valueFrom is
       // specified.
-      val stepInputs = step.inputs.map(evaluateStepInput(_, cwlEnv)).toMap
+      val stepInputs = inputs.getOrElse(step.inputs.map(evaluateStepInput(_, cwlEnv)).toMap)
```

Comment:
For scatter, the actual `stepInputs` is first evaluated as `allStepInputs`. https://github.com/dnanexus/dxCompiler/blob/1d0c55a4a624e84b35d552c9107b3fe9a61d4475/executorCwl/src/main/scala/dx/executor/cwl/CwlWorkflowExecutor.scala#L752-L758
So add an optional arg here to override.